ql/src/java/org/apache/hadoop/hive/ql/udf/generic/InstantDateTimeFormatter.java

```diff
@@ -25,15 +25,20 @@
 import java.time.LocalDate;
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
+import java.time.chrono.IsoEra;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.ResolverStyle;
+import java.time.temporal.ChronoField;
 import java.util.Objects;
 
 final class InstantDateTimeFormatter extends InstantFormatterCache<DateTimeFormatter> {
 
   InstantDateTimeFormatter(final ZoneId zoneId) {
     super(zoneId,
-        s -> new DateTimeFormatterBuilder().parseCaseInsensitive().appendPattern(s).toFormatter().withZone(zoneId));
+        s -> new DateTimeFormatterBuilder().parseCaseInsensitive()
+            .parseDefaulting(ChronoField.ERA, IsoEra.CE.getValue())
```

Comment:
Why are we setting defaults? This is the first time that I see this method been used in Hive repo. Is there a risk of creating inconsistencies with other parts of the code?

Reply:
> Why are we setting defaults?

In the new commit, this is reverted. 


The original reason for this was  according to [java docs](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html), the symbols: 
-   ` u`   is    `year  `
-    `y `   is    `year-of-era ` 

So, when resolver style is set to Strict the patten `yyyy`  is evaluated as year-of-era and would return null unless the era was specified. Hence ChronoField.ERA was set to "AD".    Without this we would be limited to using the 'uuuu' pattern to represent year .

Reply:
I see your point and I guess it makes sense but if we don't do this consistently in other places as well we may bump into some weird behavior.